gdx/src/com/badlogic/gdx/utils/compression/lzma/Encoder.java

```diff
@@ -1049,8 +1048,7 @@ public void CodeOneBlock (long[] inSize, long[] outSize, boolean[] finished) thr
 						}
 					}
 					int distance = pos;
-					for (int i = Base.kNumRepDistances - 1; i >= 1; i--)
-						_repDistances[i] = _repDistances[i - 1];
+					System.arraycopy(_repDistances, 0, _repDistances, 1, Base.kNumRepDistances - 1);
 					_repDistances[0] = distance;

```

Comment:
As said above, I'd rather not change this class, because the math in these loops looks fragile and ancient.